### DISCUSSION TITLE
Copilot does not work with Emacs Emulation in Visual Studio 2022

### DISCUSSION THREAD

**Discussion by dtecta** [community/community#56389] [Copilot Conversations] [answered]
### Select Topic Area

Question

### Body

I tried making Github Copilot work Emacs Emulation 6 in Visual Studio 2022 version 17.6.1. The problem is that Copilot expects Tab to accept a suggestion, however Tab does not interact with Copilot when Emacs Emulation is active. I've tried binding a new key to "Edit.InsertTab". However, this does not seem to have an effect. "Automatically indent when I type a tab" is disabled in C++ formatting. Is there a fix or workaround for this problem (other than disabling emacs emulation)?

**Comment by Akash1134** [comment]
Hi @dtecta ,

As far as I know, Copilot does not support Emacs Emulation in VS. I've asked my team internally for an update. 

Thanks!

**Comment by dtecta** [comment]
I

**Comment by dtecta** [comment, marked as answer]
Raised this issue in Emacs Emulation community, and got this perfect solution: https://github.com/zbrad/EmacsKeys/issues/21#issuecomment-1564011060
